### DISCUSSION TITLE
QField and Postgres

### DISCUSSION THREAD

**Discussion by philipbaileynar** [Riverscapes/QRiS#38] [Ideas & Feature Requests] [answered]
After several failed attempts, I finally got Postgres data offline editing in QField and then synchronising afterwards. 

It's fiddly and ugly. QField crashed a few times. And there's a magic step required, whereby you have to store the Postgres connection credentials as plain text in the QGIS project. So it's important we always use very limited database roles for field data entry.

Although QField feels like version 1 of a product that will be decent by version 5... this is the way forward. No more GeoPackages and ShapeFiles. Instead we get back from field work and sync directly to postgres.

**Comment by joewheaton** [comment, marked as answer]
PostGIS was tested and experimented with. It was abandoned in favor of local geopackages, with an assumption that we will do something like the following for cloud/warehouse based sharing:
- Package up specific DCEs as projects (e.g. myBRAT, myRIM, etc.) and share those to warehouse through upload ↑ with API access. Groups can be organized and shared as collections.
- If updates to a QRiS project ↓ need to be made from information in the warehouse cloud, or collected in Riverscapes Field Collector or other Apps, it will be done with an API call from QRiS to those applications.
- If we use QField ↔ QRiS, we just rely on its syncing capabilities.

We assume that users can share a zip file of the local project.  @philipbaileynar please update if I have misrepresented.
